drivers/watchdog/wdt_cc23x0.c

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (c) 2024 BayLibre, SAS
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_cc23x0_wdt
+
+#include <zephyr/drivers/watchdog.h>
+#include <zephyr/irq.h>
+#include <soc.h>
+#include <errno.h>
+
+#include <driverlib/pmctl.h>
+#include <inc/hw_ckmd.h>
+#include <inc/hw_types.h>
+#include <inc/hw_memmap.h>
+
+#define LOG_LEVEL CONFIG_WDT_LOG_LEVEL
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(wdt_cc23x0);
+
+#define CC23X0_WDT_UNLOCK(_base)        (HWREG((_base) + CKMD_O_WDTLOCK) = 0x1ACCE551)
+#define CC23X0_WDT_LOCK(_base)          (HWREG((_base) + CKMD_O_WDTLOCK) = 0x1)
+#define CC23X0_WDT_FEED(_base, _value)  (HWREG((_base) + CKMD_O_WDTCNT) = (_value))
+#define CC23X0_WDT_STALL_ENABLE(_base)  (HWREG((_base) + CKMD_O_WDTTEST) = 0x1)
+#define CC23X0_WDT_STALL_DISABLE(_base) (HWREG((_base) + CKMD_O_WDTTEST) = 0x0)
+
+#define WDT_SOURCE_FREQ      32768
+#define WDT_MS_RATIO         (WDT_SOURCE_FREQ / 1000)
+#define WDT_MAX_RELOAD_MS    0xffffffff / WDT_MS_RATIO
+#define WDT_MS_TO_TICKS(_ms) ((_ms) * WDT_MS_RATIO)
+
+struct wdt_cc23x0_data {
+	uint8_t enabled;
+	uint32_t reload;
+	uint8_t flags;
+};
+
+struct wdt_cc23x0_config {
+	uint32_t base;
+};
+
+static int wdt_cc23x0_install_timeout(const struct device *dev, const struct wdt_timeout_cfg *cfg)
+{
+	struct wdt_cc23x0_data *data = dev->data;
+
+	/* window watchdog not supported */
+	if (cfg->window.min != 0U || cfg->window.max == 0U) {
+		return -EINVAL;
+	}
+
+	if (cfg->window.max > WDT_MAX_RELOAD_MS) {
+		return -EINVAL;
+	}
+	data->reload = WDT_MS_TO_TICKS(cfg->window.max);
+	data->flags = cfg->flags;
+
+	LOG_DBG("raw reload value: %d", data->reload);
+
+	return 0;
+}
+
+static int wdt_cc23x0_setup(const struct device *dev, uint8_t options)
+{
+	const struct wdt_cc23x0_config *config = dev->config;
+
+	/* Stall the WDT counter when halted by debugger */
+	if (options & WDT_OPT_PAUSE_HALTED_BY_DBG) {
+		CC23X0_WDT_STALL_ENABLE(config->base);
+	} else {
+		CC23X0_WDT_STALL_DISABLE(config->base);
+	}
+
+	return 0;
+}
+
+static int wdt_cc23x0_disable(const struct device *dev)
+{
+	return -ENOTSUP;
+}
+
+static int wdt_cc23x0_feed(const struct device *dev, int channel_id)
+{
+	const struct wdt_cc23x0_config *config = dev->config;
+	struct wdt_cc23x0_data *data = dev->data;
+
+	CC23X0_WDT_UNLOCK(config->base);
+	CC23X0_WDT_FEED(config->base, data->reload);
+
+	return 0;
+}
+
+#ifdef CONFIG_DEBUG
+static void wdt_cc23x0_boot_reason(void)
```

Comment:
There is the `hwinfo_get_reset_cause` API for this purpose, so I'd suggest to remove this function from the driver, even if it's only enabled in debug mode.

Reply:
Thanks!
I moved it to hwinfo here is [PR](https://github.com/zephyrproject-rtos/zephyr/pull/89508)